tests/Bridge/Laravel/Http/Middlewares/AuditMiddlewareTest.php

```diff
@@ -0,0 +1,259 @@
+     */
+    public function testHandleHandlesExceptionsAndRethrows(): void
+    {
+        $this->expectException(InvalidDocumentClassException::class);
+        $this->expectExceptionMessage('Something failed.');
+
+        $middleware = $this->getMiddleware();
+
+        $middleware->handle(
+            $this->getRequest(),
+            static function (): void {
+                throw new InvalidDocumentClassException('Something failed.');
+            }
+        );
+    }
+
+    /**
+     * Test if handle can intercept response which is already in PSR7 format
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHandleWhenResponseFromControllerIsAlreadyPsr7(): void
+    {
+        $middleware = $this->getMiddleware();
+
+        $response = $middleware->handle(
+            $this->getRequest(),
+            static function (): ResponseInterface {
+                return new TextResponse('OK');
+            }
+        );
+
+        self::assertInstanceOf(ResponseInterface::class, $response);
+        self::assertSame('OK', (string)$response->getBody());
+    }
+
+    /**
+     * Test if exception thrown by http logger service is logged using logger interface
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHttpLoggerExceptionsAreLoggedByLoggerInterface(): void
+    {
+        $httpLogger = new HttpLoggerExceptionStub(); // this stub specifically throws an exception
+        $logHandler = new LogHandlerStub();
+        $middleware = $this->getMiddleware($httpLogger, $logHandler);
+
+        $middleware->handle(
+            $this->getRequest(),
+            static function (): Response {
+                return new Response('OK');
+            }
+        );
+
+        self::assertCount(1, $logHandler->getLogs());
+        $log = $logHandler->getLogs()[0];
+        self::assertArrayHasKey('message', $log);
+    }
+
+    /**
+     * Test to assert that an exception is caught and the process
+     * continues to generate psr7 request/response
+     *
+     * @return void
+     */
+    public function testProcessingContinuesWhenExceptionIsThrownInOriginalProcess(): void
+    {
+        $httpLogger = new HttpLoggerStub();
+        $middleware = $this->getMiddleware($httpLogger);
+
+        try {
+            $middleware->handle(
+                $this->getRequest(),
+                static function (): void {
+                    throw new InvalidDocumentClassException('Something failed.');
+                }
+            );
+        } catch (Exception $exception) {
+            // catching exception so we can assert the the process continues to send data to http logger
+        }
+
+        self::assertInstanceOf(ServerRequestInterface::class, $httpLogger->getRequest());
+        self::assertNull($httpLogger->getResponse());
```

Comment:
Null, or empty string?

Reply:
PSR response in HttpLogger is nullable. It can be empty string but that can be debated I think. I discussed this with Tim and the thought was that if the response is not available or could not be built for any reason (exceptions), it can be nullable.

Reply:
👍